plugins/BEdita/Core/src/Model/Action/ListRelatedObjectsAction.php

```diff
@@ -29,16 +35,86 @@ class ListRelatedObjectsAction extends ListAssociatedAction
     public function execute(array $data = [])
     {
         $data += ['joinData' => true];
-        $result = parent::execute($data);
-        if (!($result instanceof Query)) {
-            return $result;
+
```

Comment:
This method is quite long, maybe split it in shorter methods?
Also in a later/future refactor, not necessarily now.